src/compiler/crystal/compiler.cr

```diff
@@ -602,9 +635,20 @@ module Crystal
         registry.initialize_all
 
         builder = LLVM::PassManagerBuilder.new
-        builder.opt_level = 3
+        case optimization_mode
+        when OptimizationMode::Level3
+          builder.opt_level = 3
+          builder.use_inliner_with_threshold = 275
+        when OptimizationMode::Level2
+          builder.opt_level = 2
+          builder.use_inliner_with_threshold = 275
+        when OptimizationMode::Level1
+          builder.opt_level = 1
+          builder.use_inliner_with_threshold = 150
```

Comment:
There is other useful option to speedup compilation:
```
builder.disable_unroll_loops = true
```
But I don't know, how to match it in `optimize_with_new_pass_manager` for newer LLVM.

Reply:
Looks like, there is need to add LLVM::PassBuilderOptions.set_loop_unrolling which should map to LLVMPassBuilderOptionsSetLoopUnrolling